hack/generator/pkg/codegen/storage/storage_type_factory.go

```diff
@@ -61,38 +82,32 @@ func (f *StorageTypeFactory) Types() (astmodel.Types, error) {
 	return f.outputTypes, nil
 }
 
+// process carries out our transformations
+// Each step reads from outputTypes and puts the results back in there
 func (f *StorageTypeFactory) process() error {
 
-	f.outputTypes = make(astmodel.Types)
+	f.outputTypes = f.referenceTypes.Copy()
 
-	// Create storage variants and stash them in outputTypes so injectConversions can find them
-	storageVariants, err := f.inputTypes.Process(f.createStorageVariant)
+	// Inject OriginalVersion() methods into all our spec types
+	modifiedSpecTypes, err := f.outputTypes.Process(f.injectOriginalVersionMethod)
 	if err != nil {
 		return err
 	}
+	f.outputTypes = f.outputTypes.OverlayWith(modifiedSpecTypes)
```

Comment:
Same question here: Previously you had `AddTypes` which ensure that there was no collision (it panics if there was). Now you don't and so a collision is silently dropped. If there's an expectation that the `modifiedSpecTypes` doesn't trample something existing in `outputTypes` then that `panic` seems to be asserting a useful property and we might not want to lose it?

Reply:
Maybe it's moot because there's not ever anything in `outputTypes` to start with anyway? Even if that's the case I think the clarity of having the assert (and risk of failure) is better in case of future refactors that added a step before this and this made it possible for there to be something in `outputTypes` when you get here.

Reply:
Let me know how you think this looks with the changes to collapse `referenceTypes` and `outputTypes` together.

Reply:
This bit looks good to me now